model/intent/record_eeg.py

```diff
@@ -68,22 +71,44 @@ def main():
     print("Get ready in {} seconds".format(wait_seconds))
     time.sleep(wait_seconds)
 
-    for i in action_dict:
-        for _ in range(sesesion_count):
+    for _ in range(session_count):
+        for i in action_dict:
             input("Ready to record action {}. Press enter to continue".format(i))
-            print("Think Action {} for {} seconds".format(i, window_seconds))
-            time.sleep(window_seconds + 1)
+            print("Think Action {} for {} seconds".format(i, window_length + window_buffer))
+            time.sleep(window_length + window_buffer)
             data = board.get_current_board_data(sampling_size)
             action_dict[i].append(data)
     record_data["action_dict"] = action_dict
 
     # save record data
     print("Saving Data")
-    with open('recorded_eeg.pkl', 'wb') as f:
+    
+    # Iterate over any existant save files, deleting them if doOverwrite is True
+    current_number = 0;
+    while(True):
+        
+        # Create the filenames that may exist in the directory, starting with record_eeg.pkl, then record_eeg1.pkl, etc.
+        current_filename = create_filename(current_number)
+            
+        if(not os.path.isfile(current_filename)):
+            break;
+        
+        if(doOverwrite):
+            os.remove(current_filename)
+        current_number += 1
```

Comment:
I think I'd recommend double-checking this code here, from the way this is written it looks like this is assuming you're deleting files with the same name but different numeric suffixes. This by itself is fine, though I think this code might fail to delete all files in the path if you remove one of the files in the middle of an active sequence. Ex. Say we have 10 files numbered 1 through 10 and you delete 5. The condition on line 93 will check if 5 is file and since it doesn't exist it will probably return true. Because of this it will break prematurely and fail to process files 6-10. This could be fixed by replacing `break` with `continue` but doing so will cause an infinite loop that you'll need to address thanks to the use of `while(True)`.

Also minor nitpick, i think you might've accidentally used a semicolon here

Reply:
Yep, I should probably account for users deleting files on their own.
It was caused by my desire to lump the code for creating a new file and deleting old ones. Now the two chunks of code operate mutually exclusively.
For creating a new file, I think iterating like this is reasonable. 
For deleting old files, the code now starts with a list of the whole directory and treats each file independently.

Yep, that semicolon. I guess it still compiled?? Python is interesting, its removed.